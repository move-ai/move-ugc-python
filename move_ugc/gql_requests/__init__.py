"""Package for gql_requests defined in Move UGC API."""
