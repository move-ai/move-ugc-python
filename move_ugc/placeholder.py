"""Placeholder moduele."""


def placeholder() -> bool:
    """Create a placeholder function for pytest.

    Returns:
        bool: True
    """
    return True
