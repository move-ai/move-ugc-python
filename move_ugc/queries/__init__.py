"""Package for queries defined in Move UGC API."""
