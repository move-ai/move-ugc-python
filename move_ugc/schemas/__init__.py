"""Schemas for the move_ugc sdk."""
