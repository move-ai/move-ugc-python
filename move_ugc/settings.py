"""Global settings to be used by the SDK."""
from functools import lru_cache

from pydantic import Field, HttpUrl
from pydantic_settings import BaseSettings


class Settings(BaseSettings):
    """Global settings to be used by the SDK."""

    ugc_endpoint_url: HttpUrl = Field(
        default="https://api.move.ai/ugc/graphql",
        description="Move UGC API endpoint URL",
        title="GraphQL endpoint URL",
    )


@lru_cache(maxsize=None)
def get_settings() -> Settings:
    """Get the global settings.

    Returns:
        Settings: Global settings.
    """
    return Settings()
