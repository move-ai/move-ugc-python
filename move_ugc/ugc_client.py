"""Move UGC SDK client."""
from move_ugc.schemas.metaclient import MetaClient
from move_ugc.services.file import FileService
from move_ugc.services.job import JobService
from move_ugc.services.take import TakeService


class MoveUgc(MetaClient):
    """Move UGC SDK client."""

    @property
    def files(self) -> FileService:
        """Get the file service.

        Returns:
            FileService: The file service.
        """
        return FileService(**self.model_dump())

    @property
    def takes(self) -> TakeService:
        """Get the take service.

        Returns:
            TakeService: The take service.
        """
        return TakeService(**self.model_dump())

    @property
    def jobs(self) -> JobService:
        """Get the job service.

        Returns:
            JobService: The job service.
        """
        return JobService(**self.model_dump())
