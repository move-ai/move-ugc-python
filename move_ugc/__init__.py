"""Root of the move_ugc package."""
from .ugc_client import MoveUgc  # noqa: WPS300, F401
