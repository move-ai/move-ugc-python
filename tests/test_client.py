"""Unit tests for the creating a client."""

import pytest

from move_ugc import MoveUgc


class TestMoveUgcClient:
    """Test client class."""

    def test_create_client(self, faker, snapshot):
        """Test creating a client.

        Args:
            faker: The faker fixture.
            snapshot: The snapshot fixture.
        """
        client = MoveUgc(api_key=faker.pystr())
        assert client == snapshot

    def test_create_without_api_key(self, snapshot):
        """Test creating a client without an API key.

        Args:
            snapshot: The snapshot fixture.
        """
        with pytest.raises(ValueError, match="api_key"):
            MoveUgc(api_key="")
