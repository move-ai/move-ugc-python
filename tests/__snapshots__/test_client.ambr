# serializer version: 1
# name: TestMoveUgcClient.test_create_client
  MoveUgc(api_key=SecretStr('**********'), endpoint_url=HttpUrl('https://pytest_invalid_endpoint_url.com/'))
# ---
